### ISSUE TITLE
EditableColumns do not save when used in Model Admin

### ISSUE THREAD

**Issue by arambalakjian** [symbiote/silverstripe-gridfieldextensions#89]
When replacing a ModelAdmin gridfield with inline editing fields, new rows are not saved (either by clicking add button or hitting enter from previous row) .

Replicate by using the following function in a ModelAdmin extension (replacing gridfield name):

```
    public function getEditForm($id = null, $fields = null) {

        $form = parent::getEditForm($id, $fields);

        if ($blockGridField = $form->Fields()->fieldByName('MemberProfileOption')) {
            $config = $blockGridField->getConfig();
            $config->removeComponentsByType('GridFieldAddNewButton');
            $config->removeComponentsByType('GridFieldPrintButton');
            $config->removeComponentsByType('GridFieldExportButton');
            $config->removeComponentsByType('GridFieldDataColumns');
            $config->addComponent(new GridFieldEditableColumns());
            $config->addComponent(new GridFieldAddNewInlineButton());
            $config->addComponents(new GridFieldOrderableRows());

            $config->getComponentByType('GridFieldEditableColumns')->setDisplayFields(array(
                'ClassName'  => function($record, $column, $grid) {
                    return new DropdownField($column, 'Type', Singleton('MemberProfileOption')->getClassTypeArray());
                },
                'Title' => function($record, $column, $grid) {
                    return new TextField($column);
                }
            ));
        }

        return $form;
    }
```

**Comment by MattyBalaam** [comment]
https://github.com/silverstripe-australia/silverstripe-gridfieldextensions/issues/32

**Comment by micschk** [comment]
@arambalakjian, @MattyBalaam, I think this component does exactly what you're looking for: https://github.com/milkyway-multimedia/ss-gridfield-utils/blob/master/code/Buttons/SaveAllButton.php

**Comment by g4b0** [comment]
I submited a PR for this issue: https://github.com/silverstripe-australia/silverstripe-gridfieldextensions/pull/184

**Comment by chillu** [comment]
Confirmed this works in 4.5, this issue can be closed.

**Comment by emteknetnz** [comment]
Closing issue as it was fixed in 4.5
